third_party/freedreno/wrap/wrap-util.c

```diff
@@ -543,6 +537,14 @@ void collect_trace_file(const char* capture_file_path)
 #endif
 	LOGD("full_path is %s", full_path);
 
+	// Could remove(full_path). However, that is susceptible to TOCTOU problems.
+	// Prefer to open the target file once and keep it open for the duration of collection.
+	int concatenated_log_fd = open(full_path, O_CREAT | O_TRUNC | O_WRONLY);
+	if (concatenated_log_fd < 0) {
+		LOGI("Failed to open `%s` to collect traces. Can't collect traces. Error: %s", full_path, strerror(concatenated_log_fd));
```

Comment:
concatenated_log_fd would be -1 if the `open` fails? i guess we should use `errno` to get the actual error number?